test/async-script-loader-spec.js

```diff
@@ -14,87 +14,146 @@ class MockedComponent extends React.Component {
   }
 }
 
-const hasScript = (URL) => {
+const getScript = (URL, toBoolean = false) => {
   const scripTags = document.getElementsByTagName("script");
   for (let i = 0; i < scripTags.length; i += 1) {
     if (scripTags[i].src.indexOf(URL) > -1) {
-      return true;
+      return toBoolean ? true : scripTags[i];
     }
   }
-  return false;
+  return toBoolean ? false : undefined;
+}
+
+const hasScript = (URL) => getScript(URL, true)
+
+const documentLoadScript = (URL) => {
+  const script = getScript(URL);
+  script.onload();
+}
+
+const documentErrorScript = (URL) => {
+  const script = getScript(URL);
+  script.onerror();
 }
 
 describe("AsyncScriptLoader", () => {
   it("should be imported successfully", () => {
     assert.isNotNull(makeAsyncScriptLoader);
   });
 
-  it("should return a component that contains the passed component", () => {
-    const URL = "http://example.com";
-    const ComponentWrapper = makeAsyncScriptLoader(MockedComponent, URL);
+  it("should return a component that contains the passed component and fire asyncScriptOnLoad", () => {
+    const URL = "http://example.com/?default=true";
```

Comment:
made each `URL` unique for each test. this helps prevent mix ups like the one cleaned up below 😺